src/core/moves/proposal/tree/NearestNeighborInterchange_nonClockProposal.cpp

```diff
@@ -80,11 +80,24 @@ double NearestNeighborInterchange_nonClockProposal::getProposalTuningParameter(
  */
 double NearestNeighborInterchange_nonClockProposal::doProposal( void )
 {
+
+    // reset flag
+    failed = false;
+    
     // Get random number generator
     RandomNumberGenerator* rng     = GLOBAL_RNG;
     
     Tree& tau = tree->getValue();
     
+    // when the tree has fewer than 4 tips, nothing should be done
+    size_t num_tips = tau.getNumberOfTips();
+    size_t num_root_children = tau.getRoot().getNumberOfChildren();
```

Comment:
This test is weird. Why not then test directly if num_tips <= 4?

Reply:
Yeah, I agree this may appear a bit weird, especially due to my original comment about 'fewer than 4 tips'. By 'fewer than 4 tips' I meant 'fewer than or equal to 3 tips', _i.e._, `num_tips <= 3`.

To me, it seems that for the non-clock tree proposals to make sense, the tree that these proposals operate on needs to have at least one node that's not a tip nor root; _i.e._, the root needs to have at least one child that's not a tip.

I prefer to check for `num_tips <= num_root_children` instead of `num_tips <= 3` because the non-clock tree, in principle, could be fully bifurcating, in which case an NNI move still makes sense to operate on a 3-tip tree. [Personally, I perceive the non-clock tree proposals as proposals on _phylograms_ (trees whose nodes have no age) instead of on _unrooted trees_.] 

In another case---that's also highly unlikely or impossible with the current implementation or usage of revbayes but is biologically reasonable so might be relevant eventually---the tree could be multificating so that the root of a tree with more than 3 tips (_e.g._, a star phylogeny) might still have no grandchildren, which wouldn't be catched by checking `num_tips <= 3`.

To sum up, I wrote this check this way just to make it a bit more generic and potentially future proof. However, I do agree that this way comes with the cost of interpretability and clarity of the code. I'm happy to change it to `num_tips <= 3` if you would prefer that @hoehna.